EpsilonAuxBMS/AuxBMSTests.adoc

```diff
@@ -14,89 +14,115 @@ Elapsed 1000ms{
-Within 100ms {
-    Expect (PRECHARGE_STATUS_CAN_MESSAGE(0x721)[DATA_INDEX])
+Send (PRECHARGE_CAN_MESSAGE(CAN_ADDRESS))
+	Elapsed TIME_FRAME{
+		Expect (PRECHARGE_CAN_MESSAGE(CAN_ADDRESS))
 }
 ----
 
-== AuxVoltage Test
+== Aux Voltage Test
 ----
-Every 100ms{
-    Expect (CURRENT_VOLTAGE_AUX_BATTERY_CAN_MESSAGE(0x721)[DATA_INDEX])
+//Start Test
+Elapsed TIME_FRAME{
+    Expect (VOLTAGE_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
 }
 ----
 
-== Contactor Charge Test
+== Charge Contactor Test
 ----
-Every 100ms{
-	Expect (CONTACTOR_CURRENT_CAN_MESSAGE(0x721)[DATA_INDEX])
+//Start Test
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
 }
 ----
 
 == Contactor State Test
 ----
-Every 100ms{
-	Expect (CONTACTOR_STATE_CAN_MESSAGE(0x721)[DATA_INDEX])
+//Start Test
+Within TIME_FRAME{
+	Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX])
 }
 ----
 
 == All Contactor On Test
 ----
-Charge input test
+//Start Test
 
-Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 001) 
-     COMMON LED ON
-      }
-Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 011) 
-    COMMON LED ON
-    CHARGE LED ON
-    }
-  Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 111) 
-    COMMON LED ON
-    CHARGE LED ON
-    DISCHARGE LED ON
-    }
-  Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 111) 
-    COMMON LED ON
-    CHARGE LED ON
-    DISCHARGE LED ON
-    }
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b001) 
+        COMMON LED ON
+}
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b011) 
+        COMMON LED ON
+        CHARGE LED ON
+}
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b111) 
+        COMMON LED ON
+        CHARGE LED ON
+        DISCHARGE LED ON
+}
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b111) 
+   	    COMMON LED ON
+   	    CHARGE LED ON
+   	    DISCHARGE LED ON
+}
 ----
 
 == Discharge Off Test
 ----
-Discharge Test
-
-Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 101) 
-     COMMON LED ON
-     CHARGE LED ON
-      }
+//Start Discharge Test
 
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b101) 
+        COMMON LED ON
+        CHARGE LED ON
+}
 ----
 
 == Safety Test
 ----
-Safety Test
+//Start Safety Test
+
+Within 5s{
+    Expect (CONTACTOR_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 0b000) 
+        COMMON LED ON
+        DISCHARGE LED ON
+}
+----
 
-Wait 5s{
-    Expect (CHARGE_CAN_MESSAGE(0x721)[DATA_INDEX] == 000) 
-     COMMON LED ON
-     DISCHARGE LED ON
+== Strobe BMS Light Test
+----
+Send(BMS_CAN_MESSAGE(CAN_ADDRESS) == 0x200)
+    Within 200ms{
+        Expect (LIGHTS_CAN_MESSAGE(0x711) == 0x20)
+}
+----
+
+== Allow Charge Test
+----
+Send(BMS_CAN_MESSAGE(CAN_ADDRESS)[DATA_INDEX] == 1)
+    Within TIME_FRAME{
+        COMMON LED ON
+        CHARGE LED ON
```

Comment:
This test is an interesting one because there are two potential cases: the Common LED was off and is going to turn on, or the Common LED is on and will just stay on. To account for both of these, you may want to change your TIME_FRAME to the length of time it would take the common led to turn on if it was off (i.e. 1 second). It would also be a good idea to have a Disallow Charge Test.